include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -899,6 +899,69 @@ struct __replace_fn
 
 inline constexpr __internal::__replace_fn replace;
 
+// [alg.reverse]
+
+namespace __internal
+{
+struct __reverse_fn
+{
+    template <typename _ExecutionPolicy, std::ranges::random_access_range _R>
+        requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> &&
+                 std::permutable<std::ranges::iterator_t<_R>> && std::ranges::sized_range<_R>
+
+    std::ranges::borrowed_iterator_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+
+        oneapi::dpl::__internal::__ranges::__pattern_reverse(__dispatch_tag, std::forward<_ExecutionPolicy>(__exec),
+                                                             __r);
+        return std::ranges::borrowed_iterator_t<_R>{std::ranges::begin(__r) + std::ranges::size(__r)};
+    }
+
+}; //__reverse_fn
+} // namespace __internal
+
+inline constexpr __internal::__reverse_fn reverse;
+
+// [alg.reverse_copy]
+
+namespace __internal
+{
+struct __reverse_copy_fn
+{
+    template <typename _ExecutionPolicy, std::ranges::random_access_range _InRange,
+              std::ranges::random_access_range _OutRange>
+        requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> &&
+                 std::ranges::sized_range<_InRange> && std::ranges::sized_range<_OutRange> &&
+                 std::indirectly_copyable<std::ranges::iterator_t<_InRange>, std::ranges::iterator_t<_OutRange>>
+
+    std::ranges::in_in_out_result<std::ranges::borrowed_iterator_t<_InRange>,
+                                  std::ranges::borrowed_iterator_t<_InRange>,
+                                  std::ranges::borrowed_iterator_t<_OutRange>>
+    operator()(_ExecutionPolicy&& __exec, _InRange&& __in_r, _OutRange&& __out_r) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+
+        using _Size = std::common_type_t<std::ranges::range_size_t<_InRange>, std::ranges::range_size_t<_OutRange>>;
+        const _Size __sz = std::ranges::min((_Size)std::ranges::size(__in_r), (_Size)std::ranges::size(__out_r));
+
+        auto __last_in = std::ranges::begin(__in_r) + std::ranges::size(__in_r);
+        auto __stop_in = std::ranges::begin(__in_r) + (std::ranges::size(__in_r) - __sz);
```

Comment:
`std::ranges::size(__in_r)` is computed three times in three lines. It is not a heavy computation, but nevertheless - why not store it locally?

Reply:
The extra calculations have been removed.  As for __stop_in, it is now calculated as `__first_in + __in_r_skipped`, which is also quite short.